services/m3dbnode/server/server.go

```diff
@@ -111,6 +117,61 @@ func Run(runOpts RunOptions) {
 		logger.Fatalf("could not connect to metrics: %v", err)
 	}
 
+	// Presence of KV server config indicates embedded etcd cluster
+	if cfg.EnvironmentConfig.SeedNode != nil {
+		// Default etcd node name to HostID
+		if cfg.EnvironmentConfig.SeedNode.Name == "" {
+			name, err := cfg.HostID.Resolve()
+			if err != nil {
+				logger.Fatal("unable to resolve HostID")
+			}
+
+			logger.Infof("setting KV server name to: %s", name)
+
+			cfg.EnvironmentConfig.SeedNode.Name = name
+		}
+
+		// Default etcd client clusters if not set already
+		clusters := cfg.EnvironmentConfig.Service.ETCDClusters
+		if len(clusters) == 0 {
+			endpoints, err := config.InitialClusterEndpoints(cfg.EnvironmentConfig.SeedNode.InitialCluster)
+
+			if err != nil {
+				logger.Fatalf("unable to create etcd clusters: %v", err)
+			}
+
+			zone := cfg.EnvironmentConfig.Service.Zone
+
+			logger.Infof("setting client etcd cluster to zone [%s], endpoints %v", zone, endpoints)
```

Comment:
nit: This is definitely a nit, but we tend to use `key=value, key=value` if we are embedding the info in the message itself.

For instance:
```go 
logger.Infof("using seed nodes etcd cluster: zone=%s, endpoints=%v", zone, endpoints)
```